overlay/tun_linux.go

```diff
@@ -265,22 +265,70 @@ func (t *tun) deviceBytes() (o [16]byte) {
 	return
 }
 
+func hasNetlinkAddr(al []*netlink.Addr, x netlink.Addr) bool {
+	for i := range al {
+		if al[i].Equal(x) {
+			return true
+		}
+	}
+	return false
+}
+
+// addIPs uses netlink to add all addresses that don't exist, then it removes ones that should not be there
+func (t *tun) addIPs(link netlink.Link) error {
+	newAddrs := make([]*netlink.Addr, len(t.vpnNetworks))
+	for i := range t.vpnNetworks {
+		//todo I wish I didn't need to stringify and re-parse
```

Comment:
Might be worth looking at `"golang.org/x/net/route"` instead of the netlink abstraction we are using today for linux

Reply:
Possibly, I hadn't dug in to see if it would work for linux yet